src/kex.c

```diff
@@ -1637,6 +1637,83 @@ kex_method_diffie_hellman_group14_sha1_key_exchange(LIBSSH2_SESSION *session,
 
 
 
+/* kex_method_diffie_hellman_group14_sha256_key_exchange
+ * Diffie-Hellman Group14 Key Exchange using SHA256
+ */
+static int
+kex_method_diffie_hellman_group14_sha256_key_exchange(LIBSSH2_SESSION *session,
+                                                      key_exchange_state_low_t
+                                                      * key_state)
+{
+    static const unsigned char p_value[256] = {
```

Comment:
Could we pull this value out group14 sha1 function to reuse it instead of redefining it twice?

Reply:
> Could we pull this value out group14 sha1 function to reuse it instead of defining it twice?
Sure. Added a generic kex_method_diffie_hellman_group14_key_exchange function that is passed the DH hash function, diffie_hellman_sha1 or diffie_hellman_sha256.